sa/sa.go

```diff
@@ -978,6 +978,23 @@ func (ssa *SQLStorageAuthority) CountPendingAuthorizations(ctx context.Context,
 	return
 }
 
+// CountPendingOrders returns the number of pending, unexpired
+// orders for the given registration.
+func (ssa *SQLStorageAuthority) CountPendingOrders(ctx context.Context, regID int64) (int, error) {
+	var count int
+	err := ssa.dbMap.SelectOne(&count,
+		`SELECT count(1) FROM orders
+		WHERE registrationID = :regID AND
+		expires > :now AND
+		status = :pending`,
```

Comment:
Right now the index on the orders table is (regID, expires). I think to make this query perform well, we need to update that to (regID, status, expires). Note that the quantized fields go first, continue field (date) goes last.

Reply:
I'd recommend just tweaking the existing AddOrders migration, since it hasn't yet been applied.

Reply:
Ok, by that logic I think we shouldn't have altered the orders table in the `20171006131843_FinalizeOrder.sql` migration.

I'm going to delete the CSR field from the original `20170731115209_AddOrders.sql` migration while I'm adding the index & remove the CSR alter from the `20171006131843_FinalizeOrder.sql` migration to match what you're saying here.

Reply:
Fixed in c32e197